### PR TITLE
feat: Add ```logout``` command

### LINKED ISSUE #3585: [FEATURE] Add logout command `python -m argilla logout`
Add a new `logout` command that will remove the stored credentials generated by the login function introduced in #3582. This command will use `ArgillaCredentials.remove` method to remove the credentials and "logout" from the configured Argilla server. 

```shell
python -m argilla logout
```

### PR DESCRIPTION
**Pull request by eshwarhs**
# Description

This PR adds the ```logout``` command that will allow to logout from Argilla server and remove the locally stored credentials.

Closes #3585 

**Type of change**

- [x] New feature (non-breaking change which adds functionality)

**How Has This Been Tested**

Tested manually in a local development environment and added tests to cover the additions and changes.

**Checklist**

- [ ] I added relevant documentation
- [x] follows the style guidelines of this project
- [x] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [x] I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

File: src/argilla/tasks/logout/__main__.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+@app.callback(help="Logout from an Argilla Server")
+def logout(
+    api_url: str = typer.Option(..., help="The URL of the Argilla Server to logout from"),
+    api_key: str = typer.Option(..., help="The API key for logging out from the Argilla Server", prompt="API Key"),
+    workspace: Optional[str] = typer.Option(
+        None, help="The default workspace over which the operations will be performed"
+    ),
+    extra_headers: Optional[str] = typer.Option(
+        None, help="A JSON string with extra headers to be sent in the requests to the Argilla Server"
+    ),
```
Comment (gabrielmbmb) [inline src/argilla/tasks/logout/__main__.py]: For the logout we don't need any option

File: src/argilla/tasks/logout/__main__.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        logout_func(api_url=api_url, api_key=api_key, workspace=workspace, extra_headers=headers)
+        typer.echo(f"Logged out successfully from '{api_url}' Argilla server!")
+    except json.JSONDecodeError as e:
+        typer.echo("The provided extra headers are not a valid JSON string.")
+        raise typer.Exit(code=1) from e
+    except ValueError as e:
+        typer.echo(
+            f"Could not logout from the '{api_url}' Argilla server. Please check the provided credentials and try again."
+        )
+        raise typer.Exit(code=1) from e
```
Comment (gabrielmbmb) [inline src/argilla/tasks/logout/__main__.py]: In the logout command we just need to:

1. Call the `init_callback` function to check that we're logged in an Argilla server https://github.com/argilla-io/argilla/blob/288df9effb5e1127f138e11d117f6a762577a4b2/src/argilla/tasks/callback.py#L21-L26
2. Call `ArgillaCredentials.remove()` and print an informative message.

File: src/argilla/client/logout.py
Comment (gabrielmbmb) [inline src/argilla/client/logout.py:1]: We don't need this module

**Review by gabrielmbmb — CHANGES_REQUESTED** [review]
@eshwarhs Thansk for your pull request! I wrote you some comments that needs to be addressed before we can merge this branch.

**Comment by davidberenstein1957** [comment]
Awesome @eshwarhs, this is really useful :)

**Review by gabrielmbmb — APPROVED** [review]

**Pull request merged**